tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/device_info.py

```diff
@@ -42,28 +70,65 @@ def model(self):
     def software(self):
         return self.marionette.find_element(*self._software_locator).text
 
+    def tap_your_rights(self):
+        element = self.marionette.find_element(*self._your_rights_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._rights_page_locator))
+
+    def tap_your_privacy(self):
+        element = self.marionette.find_element(*self._your_privacy_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._privacy_page_header_locator))
+
+    def tap_legal_info(self):
+        element = self.marionette.find_element(*self._legal_info_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._open_source_notices_locator))
+
+    def tap_open_source_notices(self):
+        element = self.marionette.find_element(*self._open_source_notices_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._notice_text_locator))
+        time.sleep(3)  # loads the HTML file, needs to wait until it completes loading the page
```

Comment:
I thought something like this could work to prevent this time.sleep:
        iframe = self.marionette.find_element(*self._notice_text_locator)
        Wait(self.marionette).until(expected.element_displayed(iframe))
        self.marionette.switch_to_frame(iframe)
        _last_child_locator = (By.CSS_SELECTOR, 'body > *:last-child')
        Wait(self.marionette).until(expected.element_displayed(*_last_child_locator))
      self.apps.switch_to_displayed_app()

But for some reason switch_to_frame() can't find the frame, it says:  NoSuchFrameException: NoSuchFrameException: Unable to locate frame: id
I don't know why that's happening.